django/templatetags/tz.py

```diff
@@ -1,13 +1,35 @@
 from datetime import datetime, tzinfo
 
-import pytz
+try:
+    import zoneinfo
+except ImportError:
+    from backports import zoneinfo
 
+from django.conf import settings
 from django.template import Library, Node, TemplateSyntaxError
 from django.utils import timezone
 
 register = Library()
 
 
+class UnknownTimezoneException(BaseException):
+    pass
+
+
+# RemovedInDjango50Warning
+def timezone_constructor(tzname):
+    if settings.USE_DEPRECATED_PYTZ:
+        import pytz
+        try:
+            return pytz.timezone(tzname)
+        except pytz.UnknownTimeZoneError:
+            raise UnknownTimezoneException()
+    try:
+        return zoneinfo.ZoneInfo(tzname)
+    except zoneinfo.ZoneInfoNotFoundError:
+        raise UnknownTimezoneException()
```

Comment:
We test that this exception is ignored (rather than raised), see `test_localtime_filters_do_not_raise_exceptions`.

Given this exception is never intended to be seen (according to the tests atleast) do we need to change the exception being raised?

Reply:
If `pytz` is not installed we can't catch it in the calling code, so I introduced the new exception as a shim.

Reply:
```suggestion
        raise UnknownTimezoneException
```